### PR TITLE
operator: add labels to Dockerfile

### PR DESCRIPTION
**Pull request by kkujawa-sumo**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,6 +1,14 @@
 # Build the manager binary
 FROM quay.io/operator-framework/helm-operator:v1.6.1
 
+LABEL name="Sumologic Kubernetes Collection Helm Operator"
+      maintainer="support@sumologic.com"
```
Comment (pmalek-sumo) [inline Dockerfile:5]: So did we agree on this actually?

File: Dockerfile [hunk 2]
```diff
@@ -1,6 +1,14 @@
...
+      vendor="sumologic"
+      version="0.0.0"
+      release="0"
```
Comment (pmalek-sumo) [inline Dockerfile:8]: * version could be injected as build arg, WDYT?
* what's the purpose of release? incremented on every release? do we really need it? if so also a build arg?

Comment (perk-sumo) [reply Dockerfile:8]: I second Patryk's question. Does the release come from the OpenShift?
What is `release` about?

File: Dockerfile [hunk 3]
```diff
@@ -1,6 +1,14 @@
 # Build the manager binary
 FROM quay.io/operator-framework/helm-operator:v1.6.1
 
+LABEL name="Sumologic Kubernetes Collection Helm Operator"
```
Comment (astencel-sumo) [inline Dockerfile:4]: There is an OCI standard for the label names here: https://github.com/opencontainers/image-spec/blob/master/annotations.md#pre-defined-annotation-keys. I propose to use those.

File: Dockerfile (hunk 2 above)
Comment (kkujawa-sumo) [reply Dockerfile:8]: `release` label indicates release number and should be incremented on every release, @deekshahegde86 am I right?

File: Dockerfile (hunk 3 above)
Comment (kkujawa-sumo) [reply Dockerfile:4]: These labels should have these keys.

File: Dockerfile (hunk 2 above)
Comment (kkujawa-sumo) [reply Dockerfile:8]: @deekshahegde86 Can labels in Dockerfile use variables defined in build args? I mean this form:
```
LABEL name="Sumologic Kubernetes Collection Helm Operator"
      maintainer="support@sumologic.com"
      vendor="sumologic"
      version=$VERSION
      release=$RELEASE_NUMBER
      summary="Sumologic Kubernetes Collection Helm Operator for the Sumo Logic Kubernetes Collection Helm Chart"
      description="Sumologic Kubernetes Collection Helm Operator deploys https://github.com/SumoLogic/sumologic-kubernetes-collection"
```
and then `$VERSION` and `$RELEASE_NUMBER` will be specified in build args e.g:
```
 docker build --build-arg VERSION="0.0.0"  --build-arg RELEASE_NUMBER="0" .
```

Comment (deekshahegde86) [reply Dockerfile:8]: @kkujawa-sumo  Yes you can.

Comment (kkujawa-sumo) [reply Dockerfile:8]: > `release` label indicates release number and should be incremented on every release

confirmed by @deekshahegde86

File: Dockerfile (hunk 1 above)
Comment (perk-sumo) [inline Dockerfile:5]:
```suggestion
      maintainer="collection@sumologic.com"
```

File: Dockerfile (hunk 3 above)
Comment (perk-sumo) [reply Dockerfile:4]: I understand those come from the Red Hat requirements.

**Review by perk-sumo — APPROVED** [review]
I'm 🆗  pending email change.

File: Dockerfile (hunk 3 above)
Comment (kkujawa-sumo) [reply Dockerfile:4]: yes

**Comment by kkujawa-sumo** [comment]
@perk-sumo all changes come from Red Hat requirements

File: Makefile [hunk 4]
```diff
@@ -5,6 +5,9 @@
 # - use environment variables to overwrite this value (e.g export VERSION=0.0.2)
 VERSION ?= 0.0.1
 
+# Helm operator release number
+RELEASE_NUMBER ?= 0
```
Comment (pmalek-sumo) [inline Makefile:9]: We should automate it

**Review by pmalek-sumo — APPROVED** [review]

File: Makefile (hunk 4 above)
Comment (kkujawa-sumo) [reply Makefile:9]: yes but in another PR

**Pull request merged**